test/integration/context-based-restrictions.v1.test.js

```diff
@@ -0,0 +1,934 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const { readExternalSources } = require('ibm-cloud-sdk-core');
+const { v4: uuidv4 } = require('uuid');
+const ContextBasedRestrictionsV1 = require('../../dist/context-based-restrictions/v1');
+
+const authHelper = require('../resources/auth-helper.js');
+// const { doesNotReject } = require('assert');
+
+// testcase timeout value (200s).
+const timeout = 200000;
+
+// Location of our config file.
+const configFile = 'context_based_restrictions_v1.env';
+
+const describe = authHelper.prepareTests(configFile);
+
+let zoneId;
+let zoneEtag;
+let ruleId;
+let ruleEtag;
+const unexistingZoneId = '648961210dab8fdffac52cc2f28e200f';
+const unexistingRuleId = '648961210dab8fdffac52cc2f28e200f';
+const invalidID = 'this_is_an_invalid_id';
+
+describe('ContextBasedRestrictionsV1_integration', () => {
+  const contextBasedRestrictionsService = ContextBasedRestrictionsV1.newInstance({});
+
+  expect(contextBasedRestrictionsService).not.toBeNull();
+
+  const config = readExternalSources(ContextBasedRestrictionsV1.DEFAULT_SERVICE_NAME);
+  const {
+    apikey: apiKey,
+    url: URL,
+    authUrl,
+    authType,
+    testAccountId: accountId,
+    testServiceName: serviceName,
+  } = config;
+
+  expect(config).not.toBeNull();
+  expect(accountId).not.toBeNull();
+  expect(accountId).toBeDefined();
+  expect(apiKey).not.toBeNull();
+  expect(apiKey).toBeDefined();
+  expect(URL).not.toBeNull();
+  expect(URL).toBeDefined();
+  expect(authUrl).not.toBeNull();
+  expect(authUrl).toBeDefined();
+  expect(authType).not.toBeNull();
+  expect(authType).toBeDefined();
+  expect(serviceName).not.toBeNull();
+  expect(serviceName).toBeDefined();
+
+  jest.setTimeout(timeout);
+
+  contextBasedRestrictionsService.enableRetries();
+
+  test('createZone() - 201', async (done) => {
+    // Request models needed by this operation.
+
+    // AddressIPAddress
+    const addressModel = {
+      type: 'ipAddress',
+      value: '169.23.56.234',
+    };
+
+    const params = {
+      name: 'an example of zone',
+      accountId,
+      addresses: [addressModel],
+      description: 'this is an example of zone',
+      transactionId: uuidv4(),
+    };
+
+    let res;
+    try {
+      res = await contextBasedRestrictionsService.createZone(params);
+    } catch (err) {
+      done(err);
+    }
+
+    expect(res).toBeDefined();
+    expect(res.status).toBe(201);
+    expect(res.result).toBeDefined();
+
+    zoneId = res.result.id;
+    zoneEtag = res.headers.etag;
+
+    done();
+  });
+
+  test('createZone() - 400 - invalid type', async () => {
```

Comment:
This test seems to be unnecessary from that viewpoint we need to exercise all used response status codes. The invalid id test exercises the `400`, and IMHO it is enough here. On the other hand, the other PRs (java and go) don't have invalid type tests (my experience is that consistency across SDKs IT tests and examples makes the maintenance easier).
What I'm trying to say is that, it is up to you if this test will be removed or not. I'm happy if it will be removed or stays here until there is another one which covers `400` for this operation.

Reply:
removed